lib/current-context.js

```diff
@@ -0,0 +1,77 @@
+// Copyright IBM Corp. 2015,2016. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+var juggler = require('loopback-datasource-juggler');
+var remoting = require('strong-remoting');
+var ClsContext = require('loopback-context-cls');
+var deprecated = require('depd')('loopback');
+
+module.exports = function(loopback) {
+
+  /**
+   * Get the current context object. The context is preserved
+   * across async calls, it behaves like a thread-local storage.
+   *
+   * @returns {ChainedContext} The context object or null.
+   */
+  loopback.getCurrentContext = function() {
+    // NOTE(bajtos) ClsContext.getCurrentContext is overriden whenever
+    // the context changes, therefore we cannot simply assign
+    // ClsContext.getCurrentContext() to loopback.getCurrentContext()
+    deprecated('loopback.getCurrentContext() is deprecated. ' +
+      'Consider using different means of passing the context around. ' +
+      'As a short-term fix, use loopback-context-cls module.');
```

Comment:
I am fine to improve the message, as long as it remains actionable. I.e. it should be clear to the user how to get rid of that deprecation warning.

IMO, your proposal @raymondfeng, describes the state but does not provide any clue to the user about how to proceed.

Which part of the message do you find confusing?